src/dune_rules/lib.mli

```diff
@@ -169,7 +169,7 @@ end
 
 (** {1 Transitive closure} *)
 
-val closure : t list -> linking:bool -> t list Resolve.Memo.t
+val closure : t list -> linking:bool -> (t * bool) list Resolve.Memo.t
```

Comment:
I suspect it may be OK to leave this function with its old signature (which will reduce the diff). The information about direct/transitive dependencies is only relevant for compilation, and for that I guess the `requires_link` function is used...